Musicca/app/src/main/java/com/example/musicca/adapters/CurrentPlaylistAdapter.java

```diff
@@ -50,83 +58,203 @@ public CurrentPlaylistAdapter(Context context, List<String> songObjectIds, Strin
+                        ex.printStackTrace();
+                    }
+                }
+            });
         }
 
-        @Override
-        public void onClick(View view) {
-            // gets item position
-            int position = getAdapterPosition();
+        // on single touch
+        // creates a new activity with the selected song, to allow user to play the song
+        private void songSelect(int position) {
             // make sure the position is valid, i.e. actually exists in the view
             if (position != RecyclerView.NO_POSITION) {
                 String songObjectId = songObjectIds.get(position);
                 // create intent for the new activity
                 Intent intent = new Intent(context, SongPlaylistActivity.class);
-                ParseQuery<Song> query = ParseQuery.getQuery(Song.class);
-                // query.setCachePolicy(ParseQuery.CachePolicy.CACHE_ELSE_NETWORK); // or CACHE_ONLY
-                query.getInBackground(songObjectId, new GetCallback<Song>() {
+                ParseQuery<Song> querySong = ParseQuery.getQuery(Song.class);
+                querySong.getInBackground(songObjectId, new GetCallback<Song>() {
                     @Override
                     public void done(Song song, com.parse.ParseException e) {
                         if (e == null) {
-                            Log.d(TAG, "song found22" + song.getTitle());
                             intent.putExtra(EXTRA_ALBUMICONURL, song.getURL());
                             intent.putExtra(EXTRA_SONGTITLE, song.getTitle());
                             intent.putExtra(EXTRA_SONGARTIST, song.getArtist());
                             intent.putExtra(EXTRA_SONGOBJECTID, song.getObjectId());
                             intent.putExtra(EXTRA_PLAYLISTOBJECTID, playlistObjectId);
                             // show the activity
                             context.startActivity(intent);
-                            Toast.makeText(context, "Song select", Toast.LENGTH_SHORT).show();
+                            Toast.makeText(context, "Song selected", Toast.LENGTH_SHORT).show();
                         } else {
-                            Log.d(TAG, "song not found22!");
+                            Toast.makeText(context, "Error showing song!", Toast.LENGTH_SHORT).show();
                         }
                     }
                 });
             }
         }
 
-        public void bind(String songObjectId) {
-            ParseQuery<Song> query = ParseQuery.getQuery(Song.class);
+        // on double tap
+        // created a new row of like in the Like class on Parse and changes the like image to a filled icon
+        public void addLike(int position) throws ParseException {
+            List<Like> likedByUser = findLikedByCurrentUser(position);
+            if (likedByUser.size() > 0) {
+                Toast.makeText(context, "song already liked!", Toast.LENGTH_SHORT).show();
+            }
```

Comment:
How about when they double tap, they can actually unlike the song? Just a suggestion, because it seemed intuitive to me to like and unlike with the same gesture.

Reply:
I agree it is more intuitive to use the same gesture to both like or unlike song

Reply:
Double tap can now either like or unlike song: https://github.com/gracewainaina/Musicca/blob/like-songs/Screenshots/doubletap_like_unlike_song.mp4